pkg/services/provisioning/dashboards/dashboard.go

```diff
@@ -3,44 +3,77 @@ package dashboards
 import (
 	"context"
 	"fmt"
-
 	"github.com/grafana/grafana/pkg/log"
+	"github.com/pkg/errors"
 )
 
-type DashboardProvisioner struct {
-	cfgReader *configReader
-	log       log.Logger
+type DashboardProvisioner interface {
+	Provision() error
+	PollChanges(ctx context.Context)
+}
+
+type DashboardProvisionerImpl struct {
+	log         log.Logger
+	fileReaders []*fileReader
 }
 
-func NewDashboardProvisioner(configDirectory string) *DashboardProvisioner {
-	log := log.New("provisioning.dashboard")
-	d := &DashboardProvisioner{
-		cfgReader: &configReader{path: configDirectory, log: log},
-		log:       log,
+func NewDashboardProvisionerImpl(configDirectory string) (*DashboardProvisionerImpl, error) {
+	logger := log.New("provisioning.dashboard")
+	cfgReader := &configReader{path: configDirectory, log: logger}
+	configs, err := cfgReader.readConfig()
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not read dashboards config")
 	}
 
-	return d
-}
+	fileReaders, err := getFileReaders(configs, logger)
 
-func (provider *DashboardProvisioner) Provision(ctx context.Context) error {
-	cfgs, err := provider.cfgReader.readConfig()
 	if err != nil {
-		return err
+		return nil, errors.Wrap(err, "Could initialize file readers")
```

Comment:
Typo: s/Could/Couldn't

Reply:
Ach my error messages seem to all over the place. Decided to rather use `Failed to` in these cases.